### ISSUE TITLE
Mapped Subnets should allow for empty mapped subnet CIDR's

### ISSUE THREAD

**Issue by jmohr01** [mataneine/terraform-provider-metanetworks#36]
Proofpoint believe it best practice to create a seperate 'mapped_subnet' without any IPv4 CIDR blocks on it, with the enterprise DNS domain, metaport connector and routing group. This tells meta that any DNS queries for the domain, query through that metaport. It does not require any CIDR blocks. After testing via the API directly, I was able to successfully create a mapped_subnet without the mapped_subnets (CIDRs). However, if done via terraform, it says it successfully creates the subnet, but it doesn't create the resource, and thus, when trying to attach a metaport, routing_group and mapped_domain resource, it fails as it does not exist. Something to do with the way the plugin handles the `mapped_subnets` attribute.

As it is set to required, I had to put an empty list in place for it to except it `mapped_subnets = []`. This allowed it to pass and 'created' it, but it did not actually create it.

If placing a null or empty string inside the list, the API throws an error. Unable to omit it due to it being required in the plugin.

TLDR, the API allows for a creating of a mapped_subnet resource without the need for specifying any CIDRs, as long as the JSON looks like below, it works. Need this same expected behaviour to happen with TF, so it allows for the creation of DNS only mapped_subnets. Proofpoint are aware that it's weird naming considering it technically isn't a mapped subnet without mapped_subnets, but this is best practice and behaviour at the moment.

JSON:
```
{
    "description": "test",
    "enabled": true,
    "mapped_subnets": [],
    "name": "API_TEST"
}
```
POST: `https://api.metanetworks.com/v1/network_elements`
